src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -557,16 +576,21 @@ exprt get_numeric_value_from_character(
   constant_exprt A_char=from_integer('A', char_type);
   constant_exprt ten_int=from_integer(10, char_type);
 
-  binary_relation_exprt upper_case(chr, ID_ge, A_char);
+  // There are four cases: lower case, upper case, a digit or '+' and '-'
   binary_relation_exprt lower_case(chr, ID_ge, a_char);
+  binary_relation_exprt upper_case_or_lower_case(chr, ID_ge, A_char);
+  binary_relation_exprt upper_case_lower_case_or_digit(chr, ID_ge, zero_char);
 
   return typecast_exprt(
     if_exprt(
       lower_case,
       plus_exprt(minus_exprt(chr, a_char), ten_int),
       if_exprt(
-        upper_case,
+        upper_case_or_lower_case,
```

Comment:
EBCDIC does appear to have them the other way around. It's also a bit bizarre. I don't think anyone uses it any more.